### PR TITLE
Added thanos rule component in the quickstart.sh

### LINKED ISSUE #2588: Add a Thanos Ruler in the quickstart.sh
## Feature Request

The [`quickstart.sh`](https://github.com/thanos-io/thanos/blob/master/scripts/quickstart.sh) is a pretty useful script for getting an example Thanos up and running. However, it would be useful if we could have a Thanos Ruler running alongside the already existing components running. This would be useful both for development and demonstration purposes.

### PR DESCRIPTION
**Pull request by yashrsharma44**
Signed-off-by: Yash <yashrsharma44@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

Fixes #2588 

## Changes

Added Thanos Rule in the `quickstart.sh`. Now users while running this script can also see the `thanos` rule in action! 

## Verification
Tested locally.

### CODE REVIEW

**Comment by yashrsharma44** [comment]
@ranjithkumar007 If you have any additions for using this `quickstart.sh` please provide so, I would really appreciate it!

**Comment by ranjithkumar007** [comment]
> @ranjithkumar007 If you have any additions for using this `quickstart.sh` please provide so, I would really appreciate it!

what about adding [compactor](https://github.com/thanos-io/thanos/blob/master/docs/quick-tutorial.md#compactor) as well to `quickstart.sh`? I am not sure though.

**Comment by bwplotka** [comment]
I think it would be nice however compactor would not do much without data

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ We use *breaking* word for marking changes that are not backward compatible (rel
 - [#2502](https://github.com/thanos-io/thanos/pull/2502) Added `hints` field to `SeriesResponse`. Hints in an opaque data structure that can be used to carry additional information from the store and its content is implementation specific.
 - [#2521](https://github.com/thanos-io/thanos/pull/2521) Sidecar: add `thanos_sidecar_reloader_reloads_failed_total`, `thanos_sidecar_reloader_reloads_total`, `thanos_sidecar_reloader_watch_errors_total`, `thanos_sidecar_reloader_watch_events_total` and `thanos_sidecar_reloader_watches` metrics.
 - [#2532](https://github.com/thanos-io/thanos/pull/2532) Store: Added hidden option for experimental caching bucket, that can cache chunks into shared memcached. This can speed up querying and reduce number of requests to object storage.
+- [#2590](https://github.com/thanos-io/thanos/pull/2590) Added thanos rule command in the `quickstart.sh` file.
```
Comment (bwplotka) [inline CHANGELOG.md:25]: This is definitely not needed in CHANGELOG (:

Comment (bwplotka) [reply CHANGELOG.md:25]: This is not interesting from perspective of operating Thanos. It only helps in Thanos development

**Comment by yashrsharma44** [comment]
I ran it in Ubuntu 16.04

File: CHANGELOG.md (hunk 1 above)
Comment (yashrsharma44) [reply CHANGELOG.md:25]: Sounds fair

**Comment by yashrsharma44** [comment]
I have rebased the changes, let me know if anything else is required!

File: scripts/quickstart.sh [hunk 2]
```diff
@@ -250,12 +263,26 @@ done
 sleep 0.5
 
 if [ -n "${GCS_BUCKET}" -o -n "${S3_ENDPOINT}" ]; then
-  ${THANOS_EXECUTABLE} tools bucket web \
+  ${THANOS_EXECUTABLE} bucket web \
```
Comment (kakkoyun) [inline scripts/quickstart.sh:266]: `bucket` command is under `tools`. This shouldn't change I believe.
Have you test it with the latest Thanos version?
```suggestion
  ${THANOS_EXECUTABLE} tools bucket web \
```

Comment (yashrsharma44) [reply scripts/quickstart.sh]: Done

**Pull request merged**